### PR TITLE
Cyphers/layout

### PR DESCRIPTION
**Pull request by diyessi**
This is the next set of tensor simplification changes, this one applying to layouts.

### CODE REVIEW

File: src/ngraph/descriptor/layout/tensor_layout.hpp [hunk 1]
```diff
@@ -35,19 +35,19 @@ namespace ngraph
             /// \brief Interface for describing implementations of tensor views.
             ///
             /// Kernel selection will need to pay attention to the layout.
-            class TensorViewLayout
+            class TensorLayout
```
Comment (aprocter) [inline src/ngraph/descriptor/layout/tensor_layout.hpp]: Suggestion for future: rename to `TensorLayoutDescriptor`

File: src/ngraph/descriptor/tensor.hpp [hunk 2]
```diff
@@ -31,7 +31,7 @@ namespace ngraph
     {
         namespace layout
         {
-            class TensorViewLayout;
+            class TensorLayout;
         }
 
         /// \brief Compile-time descriptor of a first-class value that is a view of a tensor.
```
Comment (aprocter) [inline src/ngraph/descriptor/tensor.hpp]: Suggestion for future: rename class to `TensorDescriptor`.

File: src/ngraph/node.cpp [hunk 3]
```diff
@@ -239,19 +239,19 @@ const Shape& Node::get_shape() const
     return get_output_shape(0);
 }
 
-shared_ptr<descriptor::TensorView> Node::get_output_tensor_view(size_t i) const
+shared_ptr<descriptor::Tensor> Node::get_output_tensor_ptr(size_t i) const
```
Comment (jbobba) [inline src/ngraph/node.cpp]: Can we call this get_output_tensor()?

File: src/ngraph/descriptor/input.cpp [hunk 4]
```diff
@@ -69,14 +69,14 @@ Tensor& Input::get_tensor()
     return m_output->get_tensor();
 }
 
-std::shared_ptr<const TensorView> Input::get_tensor_view() const
+std::shared_ptr<const Tensor> Input::get_tensor_ptr() const
```
Comment (jbobba) [inline src/ngraph/descriptor/input.cpp]: minor nit: can we call this get_tensor()?

File: src/ngraph/descriptor/output.hpp [hunk 5]
```diff
@@ -44,8 +44,8 @@ namespace ngraph
 
             std::shared_ptr<Node> get_node() const;
             size_t get_index() const { return m_index; }
-            std::shared_ptr<Tensor> get_tensor_view() const { return m_tensor; }
-            void set_tensor_view(const std::shared_ptr<Tensor>& tensor) { m_tensor = tensor; }
+            std::shared_ptr<Tensor> get_tensor_ptr() const { return m_tensor; }
```
Comment (jbobba) [inline src/ngraph/descriptor/output.hpp]: get_tensor() and set_tensor()?

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 6]
```diff
@@ -680,14 +680,14 @@ using namespace ngraph::runtime;
         {
             for (size_t i = 0; i < param->get_output_size(); ++i)
             {
-                shared_ptr<descriptor::TensorView> tv = param->get_output_tensor_view(i);
+                shared_ptr<descriptor::TensorView> tv = param->get_output_tensor_ptr(i);
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: Cant we switch all descriptor::TensorView to descriptor::Tensor?

File: src/ngraph/descriptor/layout/dense_tensor_layout.hpp [hunk 7]
```diff
@@ -32,23 +32,19 @@ namespace ngraph
...
+                ~DenseTensorLayout() override {}
+                DenseTensorLayout(const Tensor& tensor);
 
-                virtual size_t get_size() override { return m_size; }
                 size_t get_offset() const { return m_offset; }
                 virtual size_t get_index_offset(const std::vector<size_t>& indices) override;
-
-                const Strides& get_strides() const override { return m_strides; }
-                virtual bool operator==(const TensorViewLayout& other) const override;
+                Strides get_strides() const override;
```
Comment (jbobba) [inline src/ngraph/descriptor/layout/dense_tensor_layout.hpp]: Why not return a reference?

**Pull request merged**